packages/node_modules/@ciscospark/common/src/one-flight.js

```diff
@@ -51,7 +51,10 @@ export default function oneFlight(...params) {
       }
 
       /* eslint no-invalid-this: [0] */
-      let flight = flights.get(this, target, key);
+      let flight;
+      if (key) {
```

Comment:
`key` should only be undefined if `keyFactory` returns `undefined`, but that doesn't mean call is not in flight. Assuming `undefined` is not a valid key (and we need to confirm that), then, really, the solution needs to fix that key rather than assuming flight exists.

note that `WeakMappedMappedMap` is a WeakMap of Maps of Maps, meaning that `undefined` needs to be a valid key for `Map`s, not necessarily `WeakMaps`. `WeakMappedMappedMap` might be expressed as `WeakMap<Object, Map<Object, Map<string, Promise>>>` in e.g. typescript.

`flights.get(this, target, key)` expands to something like

```js
const targets = flights.get(this);
const flightsForTarget = targets.get(target);
const flight = flightsForTarget.get(key);
```